### PR TITLE
fix upgrade

### LINKED ISSUE #49: Error during upgrade, $tmpdir not defined in some cases
just to report https://forum.yunohost.org/t/impossible-de-mettre-a-jour-piwigo-from-2-10-1-ynh1-to-2-10-1-ynh2-cant-upgrade-piwigo-from-2-10-1-ynh1-to-2-10-1-ynh2/10531

Indeed, line 162 of the upgrade script, tmpdir seems only set previously in a `if` block

### PR DESCRIPTION
**Pull request by kay0u**
## Problem
- *#49*

## Solution
- *Fix #49*
- *Backward compatibility only when changing the version of the application*

## PR Status
- [ ] Code finished.
- [ ] Tested with Package_check.
- [ ] Fix or enhancement tested.
- [ ] Upgrade from last version tested.
- [ ] Can be reviewed and tested.

## Validation
---
*Minor decision*
- **Upgrade previous version** : 
- [x] **Code review** : Maniack C
- [x] **Approval (LGTM)** : Maniack C
- [ ] **Approval (LGTM)** : 
- **CI succeeded** : 
[Build Status](https://ci-apps-hq.yunohost.org/jenkins/job/piwigo_ynh%20PR50/)  
When the PR is marked as ready to merge, you have to wait for 3 days before really merging it.

### CODE REVIEW

**Comment by JimboJoe** [comment]
Upgrade from fresh ~ynh1 fails on my side (log [here](https://paste.yunohost.org/raw/uzidiyorij)). Did you test it on your side?

**Comment by JimboJoe** [comment]
:up:

**Comment by vrubiolo** [comment]
Hi guys, any news on the matter here? Note that the update fails for me too and for others as well, cf https://forum.yunohost.org/t/official-app-piwigo/3999/22.
Thanks much for your answer!

**Comment by JimboJoe** [comment]
Hello,
I'm sorry, but bumping everywhere and everyday unfortunately won't make things happen... I personally don't have time at the moment, please step in to help fixing or wait for other contributors to do so...

**Comment by vrubiolo** [comment]
Hi @JimboJoe 

Thanks for replying and sorry for appearing to be pushy. That was not my intent.

There seemed to be no activity on any of the threads so I was wondering what was going on after you asked for testing feedback on https://forum.yunohost.org/t/official-app-piwigo/3999/23?u=vinz and that me and mr_smithers had replied with our results.

I perfectly understand you might not have time for now. Thanks for saying so. That allows others to try and work on it. The Yuno team appears to be made entirely of volunteers who work on their free time so this is perfectly normal.

**Comment by JimboJoe** [comment]
:up: 
Several reports on the forum that that fix eventually works... :sweat_smile:

**Comment by kay0u** [comment]
I can't approve my own PR :-)

**Comment by JimboJoe** [comment]
At least you can verbally approve my latest changes :wink:

File: scripts/upgrade [hunk 1]
```diff
@@ -31,6 +31,9 @@ admin_pwd=$(ynh_app_setting_get --app=$app --key=admin_pwd)
 #=================================================
 
 upgrade_type=$(ynh_check_app_version_changed)
+# Workaround for now to force all instances to get the latest LDAP plugin
+# in order to create the dedicated tables
+upgrade_type="UPGRADE_APP"
```
Comment (kay0u) [inline scripts/upgrade]: I'd surround this line by an `if the current version is the problematic version` but it's not really important

Comment (JimboJoe) [reply scripts/upgrade]: Indeed, but I will remove it when rebasing the 2.10.2 update PR that came at the right time!

**Comment by JimboJoe** [comment]
Let's say we can merge in 3 days.

**Pull request merged**